tests/pytest_tests/unit_tests/test_launch/test_controller/test_base.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import MagicMock
+
+import pytest
+from wandb.sdk.launch.agent2.controllers.base import BaseManager
+from wandb.sdk.launch.agent2.jobset import JobSet
+from wandb.sdk.launch.queue_driver.abstract import AbstractQueueDriver
+
+
+class TestBaseManager(BaseManager):
+    resource_type = "test"
+
+    def __init__(self, config, jobset, logger, legacy, max_concurrency):
+        self.queue_driver = MagicMock(spec=AbstractQueueDriver)
+        super().__init__(config, jobset, logger, legacy, max_concurrency)
+
+
+@pytest.fixture
+def jobset():
+    return MagicMock(spec=JobSet)
+
+
+@pytest.fixture
+def config():
+    return {"jobset_spec": {"name": "test"}, "agent_id": "test"}
+
+
+def test_reconcile_launch_item(jobset):
+    mgr = TestBaseManager()
```

Comment:
If we want pure unit tests, we'll want to add a separate `test_launch_item`, and mock out `BaseManager.launch_item()` in this test. Likewise, we'll want a `test_cancel_job` and maybe a `test_release_item` and `test_pop_next_item`.